src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php

```diff
@@ -6,16 +6,16 @@
 use Illuminate\Support\Str;
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Relations\Relation;
 use Illuminate\Database\Eloquent\Relations\HasOne;
-use Illuminate\Database\Eloquent\Relations\HasMany;
-use Illuminate\Database\Eloquent\Relations\MorphTo;
 use Illuminate\Database\Eloquent\Relations\MorphOne;
-use Illuminate\Database\Eloquent\Relations\Relation;
 use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Relations\MorphTo;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Eloquent\Relations\HasManyThrough;
 use Illuminate\Database\Eloquent\Relations\MorphMany;
-use Illuminate\Database\Eloquent\Relations\MorphToMany;
 use Illuminate\Database\Eloquent\Relations\BelongsToMany;
-use Illuminate\Database\Eloquent\Relations\HasManyThrough;
+use Illuminate\Database\Eloquent\Relations\MorphToMany;
```

Comment:
Imports should be sorted by length. Helps review the PR.

Reply:
@deleugpn I am not sure what are you mean.
1.I can't create new relation instance inside new newRelation method because each relation has different arguments and the completely different way how the instance is created.

I only can do something like this : 

    protected static $relationClasses = [
        HasOne::class => HasOne::class,
        MorphOne::class => MorphOne::class,
        BelongsTo::class => BelongsTo::class,
        BelongsTo::class => MorphTo::class,
        HasMany::class => HasMany::class,
        HasManyThrough::class => HasManyThrough::class,
        MorphMany::class => MorphMany::class,
        BelongsToMany::class => BelongsToMany::class,
        MorphToMany::class => MorphToMany::class,
    ];

    protected static function relationClass($relation){
        return self::$relationClasses[$relation];
    }

    public function hasOne($related, $foreignKey = null, $localKey = null)
    {
        $instance = $this->newRelatedInstance($related);

        $foreignKey = $foreignKey ?: $this->getForeignKey();

        $localKey = $localKey ?: $this->getKeyName();

        $relationClass = static::relationClass(HasOne::class);

        return new $relationClass($instance->newQuery(), $this, $instance->getTable().'.'.$foreignKey, $localKey);
    }